examples/modules/fetch-newrelic-api-access-key/main.tf

```diff
@@ -0,0 +1,41 @@
+
+provider "graphql" {
+  url = var.fetch_access_keys_service.graphiql_url
+  headers = {
+    "Content-Type" = "application/json"
+    "API-Key" = var.fetch_access_keys_service.api_key != "" ? var.fetch_access_keys_service.api_key : var.create_access_keys_service.api_key
+  }
+}
+
+data "graphql_query" "basic_query" {
+  query_variables = {
+    "id"        = var.fetch_access_keys_service.key_id
+    "key_type"  = var.fetch_access_keys_service.key_type
+  }
+  query  = file("query.gql")
+  count = local.is_resource_created ? 0 : 1
+}
+
+resource "newrelic_api_access_key" "api_access_key" {
+  count  = var.create_access_keys_service.newrelic_account_id != "" ? 1 : 0
+  account_id  = var.create_access_keys_service.newrelic_account_id
+  key_type    = var.create_access_keys_service.key_type
```

Comment:
@vinay-newrelic upon testing I find that this module works fine to create `INGEST` - `LICENSE` and `INGEST` - `BROWSER` keys but not USER keys. 

<img width="1263" alt="image" src="https://github.com/user-attachments/assets/325712f5-2901-424b-ae7e-aaa229e8b002">

This is because we need the attribute `user_id` here (coming in as an input from the module) as the attribute `user_id` is required to create a USER API Key. You might want to add `user_id` as an optional attribute in the create module and pass it to the `newrelic_api_access_key` resource.

Reply:
@pranav-new-relic my bad i missed this key while creating api_access key ..I have now added this an conditional field.So it would be required when key_type == USER and optional in other cases.